component/pyroscope/ebpf/ebpfspy/session.go

```diff
@@ -0,0 +1,313 @@
+//go:build linux
+
+// Package ebpfspy provides integration with Linux eBPF. It is a rough copy of profile.py from BCC tools:
+//
+//	https://github.com/iovisor/bcc/blob/master/tools/profile.py
+package ebpfspy
+
+import (
+	_ "embed"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pyroscope-io/pyroscope/pkg/agent/ebpfspy/cpuonline"
```

Comment:
I noticed this dependency because Pyroscope is AGPLv3 (https://github.com/grafana/pyroscope/blob/v0.37.2/LICENSE). Isn't this incompatible to Apache 2? I guess cpuonline is not that large and can be moved into the agent repo?

Reply:
Yes I agree either change back the license of `cpuonline` package to Apache 2.0 or move the code here.

Reply:
moved it here